### PR TITLE
DOC:sparse.linalg: Update splu and spilu docstrings

### LINKED ISSUE #8140: sparse LU decomposition does not solve with complex right-hand side
The `solve` method of the sparse LU-decomposition `splu` or `spilu` throws a `TypeError` if called with a `numpy.array` of type `numpy.complex`. I am actually using `spilu` for preconditioning a gmres-solver required to perform a linear solve in a non-symmetric generalized eigenvalue problem. The eigenvectors are complex and hence the right-hand side for the linear solve can be complex.

### Reproducing code example:
```
import numpy as np
from scipy.sparse import csr_matrix
import scipy.sparse.linalg as sp_sparse_la

A = csr_matrix([[2.,-1.],[-1.,2.]])
n = A.shape[0]

v_real = np.random.randn(n)
v_cmplx = np.random.randn(n) + 1.0J * np.random.randn(n)

luA = sp_sparse_la.splu(A)
x_real = luA.solve(v_real)
x_cmplx = luA.solve(v_cmplx)
```

### Error message:
```
Traceback (most recent call last):
  File "dump.py", line 20, in <module>
    x_cmplx = luA.solve(v_cmplx)
TypeError: Cannot cast array data from dtype('complex128') to dtype('float64') according to the rule 'safe'
```

### Scipy/Numpy/Python version information:
```
('1.0.0', '1.13.3', sys.version_info(major=2, minor=7, micro=12, releaselevel='final', serial=0))
```

### PR DESCRIPTION
**Pull request by chachaleo**
Adding comment on the possible need for the matrix A to be complex, if RHS is complex for solve

#### Reference issue
Closes #8140

### CODE REVIEW

File: scipy/sparse/linalg/dsolve/linsolve.py [hunk 1]
```diff
@@ -287,6 +287,9 @@ def splu(A, permc_spec=None, diag_pivot_thresh=None,
     Notes
     -----
     This function uses the SuperLU library.
+    
+    In the case of using scipy.sparse.linalg.splu as a preconditionner for eignenvalue problem
+    you migth need to cast A as dtype('complex128') as eigenvalue could be complex.
```
Comment (tylerjereddy) [inline scipy/sparse/linalg/dsolve/linsolve.py:292]: typos for:

- `eignenvalue`
- `migth`

**Review by tylerjereddy — COMMENTED** [review]
I don't know if there was a discussion somewhere about the need for this--I'll defer to an expert on sparse/linalg stuff.

**Comment by ilayn** [comment]
Can you please describe what the statement is mentioning? I don't think I understand the intention

**Comment by chachaleo** [comment]
It is for the issue sparse LU decomposition does solve with complex right-hand side #8140

**Comment by leofang** [comment]
So can it be complex64?

**Comment by ilayn** [comment]
The issue here is that when solving sparse `Ax = b`, the solver picks up the dtype flavor only by looking at `A`. Hence if `b` is complex then it throws an error because the underlying algorithm expects a real `b` and that is the problem that is addressed in #8140 

https://github.com/scipy/scipy/blob/701ffcc8a6f04509d115aac5e5681c538b5265a2/scipy/sparse/linalg/dsolve/_superluobject.c#L85-L87

We want to mention this in docs hence the PR but it requires some tweaks. I think something along the lines of

```
When a real array is factorized and the returned SuperLU object "solve()" method is used with complex arguments an error is generated. Instead cast the initial matrix to complex and then factorize. 
```

or something like that

**Comment by lucascolley** [comment]
Hey @chachaleo, would you be interested in returning to this? I'm happy to review if you would like to finish! Looks like there are just a few comments to address above.

**Comment by j-bowhay** [comment]
I am going to close this for now as there has been no activity

**Pull request closed without merging**